DocumentsFromSnapshotMigration/src/main/java/com/rfs/RfsMigrateDocuments.java

```diff
@@ -120,8 +155,16 @@ public static void main(String[] args) throws Exception {
                         new OpenSearchClient(arguments.targetHost, arguments.targetUser, arguments.targetPass, false);
                 DocumentReindexer reindexer = new DocumentReindexer(targetClient);
 
-                SourceRepo sourceRepo = S3Repo.create(Paths.get(arguments.s3LocalDirPath),
-                        new S3Uri(arguments.s3RepoUri), arguments.s3Region);
+                SourceRepo sourceRepo;
+                if (arguments.snapshotLocalDirPath == null) {
+                    sourceRepo = S3Repo.create(
+                        Paths.get(arguments.s3LocalDirPath),
+                        new S3Uri(arguments.s3RepoUri),
+                        arguments.s3Region
+                    );
+                } else {
+                    sourceRepo = new FileSystemRepo(Paths.get(arguments.snapshotLocalDirPath));
```

Comment:
lift the args check up and add a default else to throw to make the code more defensible for the future.